src/toil_lib/tools/preprocessing.py

```diff
@@ -82,221 +84,425 @@ def run_samtools_index(job, bam_id):
+    :param bool unsafe: If True, runs GATK tools in UNSAFE mode: "-U ALLOW_SEQ_DICT_INCOMPATIBILITY"
+    :return: FileStoreIDs for BAM and BAI files
     :rtype: tuple(str, str)
     """
-    rtc = job.wrapJobFn(run_realigner_target_creator, bam, bai, ref, ref_dict,
-                        fai, phase, mills, mem, unsafe, cores=job.cores)
-    ir = job.wrapJobFn(run_indel_realignment, rtc.rv(), bam, bai, ref, ref_dict,
-                       fai, phase, mills, mem, unsafe, cores=job.cores)
-    br = job.wrapJobFn(run_base_recalibration, ir.rv(0), ir.rv(1), ref, ref_dict,
-                       fai, dbsnp, mem, unsafe, cores=job.cores)
-    pr = job.wrapJobFn(run_print_reads, br.rv(), ir.rv(0), ir.rv(1), ref, ref_dict,
-                       fai, mem, unsafe, cores=job.cores)
-    # Wiring
-    job.addChild(rtc)
-    rtc.addChild(ir)
-    ir.addChild(br)
-    br.addChild(pr)
-    return pr.rv(0), pr.rv(1)
-
-
-def run_realigner_target_creator(job, bam, bai, ref, ref_dict, fai, phase, mills, mem, unsafe=False):
+    # MarkDuplicates runs best when Xmx <= 10G
+    mdups_memory = min(10737418240, job.memory)
+
+    # The MarkDuplicates disk requirement depends on the input BAM and BAI files and the output
+    # BAM and BAI files. The output BAM file is approximately the same size as the input BAM file.
+    mdups_disk = PromisedRequirement(lambda bam_, bai_: 2 * (bam_.size + bai_.size), bam, bai)
+    mdups = job.wrapJobFn(picard_mark_duplicates,
+                          bam,
+                          bai,
+                          cores=job.cores,
+                          disk=mdups_disk,
+                          memory=mdups_memory)
+
+    # Get genome reference file sizes for calculating disk requirements
+    genome_ref_size = ref.size + ref_dict.size + fai.size
+
+    # Get INDEL resource file sizes and genome reference file sizes
+    indel_ref_size = mills.size + g1k.size + genome_ref_size
+
+    # The RealignerTargetCreator disk requirement depends on the input BAM/BAI files, the genome reference files, and
+    # the output intervals file. The intervals file size is less than the reference file size, so estimate the interval
+    # file size as the reference file size.
+    realigner_target_disk = PromisedRequirement(lambda bam_, bai_, ref_size:
+                                                bam_.size + bai_.size + 2 * ref_size,
+                                                mdups.rv(0),
+                                                mdups.rv(1),
+                                                indel_ref_size)
+
+    realigner_target = job.wrapJobFn(run_realigner_target_creator,
+                                     mdups.rv(0),
+                                     mdups.rv(1),
+                                     ref, ref_dict, fai,
+                                     g1k, mills,
+                                     unsafe=unsafe,
+                                     cores=job.cores,
```

Comment:
I can't remember which one it is, but one (both?) of the realigner target creator and the actual realigner are single threaded, IIRC. Those shouldn't be passed a cores=job.cores, they should be passed a cores=1, IMO.

Thoughts @jvivian @hannes-ucsc ? IIRC, picard is the same way.